CMakeLists.txt

```diff
@@ -101,12 +101,10 @@ file(RELATIVE_PATH
   "${CMAKE_INSTALL_PREFIX}"
 )
 
-# TODO(jrivero): CMake 3.20 provides cmake_path(APPEND) which implements the
-# same functionality as join_paths(). Remove JoinPaths in the next major version
-# if all the supported platforms are in 3.20 version.
-include(JoinPaths)
-join_paths(GZ_PC_LIBDIR "\${prefix}" "${CMAKE_INSTALL_LIBDIR}")
-join_paths(GZ_PC_INCLUDEDIR "\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}" "${GZ_INCLUDE_INSTALL_DIR_POSTFIX}")
+cmake_path(APPEND "\${prefix}" "${CMAKE_INSTALL_LIBDIR}"
+    OUTPUT_VARIABLE GZ_PC_LIBDIR)
+cmake_path(APPEND "\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}" "${GZ_INCLUDE_INSTALL_DIR_POSTFIX}"
+    OUTPUT_VARIABLE GZ_PC_INCLUDEDIR)
```

Comment:
this doesn't actually work the same

before this change gz-cmake4.pc has the following line:

~~~
includedir=${prefix}/include/gz/cmake4
~~~

and after this change it is

~~~
includedir=include/gz/cmake4
~~~

I think the first argument to `cmake_path(APPEND` needs to be a cmake variable, not an expression that is meant to be evaluated at pkg-config runtime. I'm not sure how to use `cmake_path(APPEND` in this case